scripts/dump_load_db.sh

```diff
@@ -0,0 +1,89 @@
+#!/bin/bash
+
+# some config variables
+OUTPUT_DIR="db_dumps"
+BASE="db-dump"
+TIMESTAMP=$(date +"%Y-%m-%d_%H%M")
+ENV_FILE=../.docker/database-variables.env
+CONTAINER="herdbook-db"
+DUMP_FORMAT="tar"
+
+if [[ ! "$1" =~ ^(load|dump)$ ]]
+then
+  cat << HELP
+USAGE: ./dump_load_db.sh [load [filename] | dump [directory]]
+
+Viable options are:
+
+      load <file> --clean       load database dump <file>. Optional flag --clean
+
+      dump <directory>          create a dump and place it in <directory>
+
+HELP
+  exit 0
+fi
+
+if [ "$( docker container inspect -f '{{.State.Status}}' ${CONTAINER} )" != "running" ]
+then
+  echo "Database container need to be running to perform backup operations"
+  exit 1
+fi
+
+# load database variables
+if [ ! -e "${ENV_FILE}" ]
+  then
+    echo "Couldn't find env file '$ENV_FILE'." >&2
+    exit 1
+  fi
+
+source ../.docker/database-variables.env
+
+FLAGS="-U $POSTGRES_USER -d $POSTGRES_DB"
+
+if [[ "$1" == "load" ]]
+then
+  if [[ -z "$2" ]]
+  then
+    echo "Please specify database dump file"
+    exit 1
+  fi
+  DUMP_FILE=$2
+  if [ ! -e "${DUMP_FILE}" ]
+  then
+    echo "Couldn't find database dump file '$DUMP_FILE'." >&2
+    exit 1
+  fi
+  if [[ "$3" == "--clean" ]]
+  then
+    echo "NOTE flag --clean will drop database objects before recreating them. Do you want to continue?(yes/no)"
+    read input
+    if [ "$input" == "yes" ]
+    then
+      FLAGS="--clean $FLAGS"
+    fi
+  fi
+  echo "Restoring database from dump: ${DUMP_FILE}"
+  cat "$DUMP_FILE" | docker exec -i "${CONTAINER}" pg_restore $FLAGS
+
+else
+  if [[ "$1" == "dump" ]]
+  then
+    if [[ -z "$2" ]]
+    then
+      echo "Please specify which directory to put dump file into"
+      exit 1
+    fi
+    OUTPUT_DIR=$2
+    if [ ! -e "$OUTPUT_DIR" ]
+    then
+      echo "Creating dump output directory"
+      mkdir $OUTPUT_DIR
+    fi
```

Comment:
As a personal preference, I like `mkdir -p SOMETHING` instead of things like these (the error message is also informative enough).